frontend/components/puzzle2.jsx

```diff
@@ -1,20 +1,46 @@
 import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
 
 // Answer: See o double you
-
 class Puzzle2 extends React.Component {
-  render() {
 
-    return (
-      <div>
+ constructor(props) {
+    super(props);
+    
+    this.state = {
+      puzzlesolved: 'no',
+      blankanswer: 'no',
+      wronganswer: 'no'
+    }
+ };
+   submitanswer() {
+      var userinput = document.getElementById('puzzleAnswer').value;
```

Comment:
I'm generally not a fan of doing `document.getElementById` in React because it depends on the HTML layout. The React way is to put an onChange listener on the input and keep track of the value in the state, then accessing the state to get the value of the element

Reply:
hmmm i see. Okay, will need to do more research.